sys/arduino/arduino_sketches.inc.mk

```diff
@@ -0,0 +1,31 @@
+# Compile together the Arduino sketches of the application
+# They are declared a as new module $(SKETCH_MODULE) in $(SKETCH_MOD_DIR)
+
+ifndef SKETCH_MODULE
+  $(error SKETCH_MODULE undefined. It should be defined to the sketches module name)
+endif
+ifndef SKETCH_MODULE_DIR
+  $(error SKETCH_MODULE_DIR undefined. It should be defined to the sketches module directory)
+endif
+ifndef SKETCHES
+  $(error SKETCHES undefined. It should be defined to the list of sketches files)
+endif
+
+SNIPDIR         = $(RIOTBASE)/sys/arduino
+SKETCHES_ALL    = $(SNIPDIR)/pre.snip $(SKETCHES) $(SNIPDIR)/post.snip
+SKETCH_CPP     ?= arduino_sketches.cpp
+
+# Building the module files
+#   Do not use $^ in receipes as Makefile is also a prerequisite
+$(SKETCH_MODULE_DIR)/Makefile: $(SKETCH_MODULE_DIR)/$(SKETCH_CPP)
+	$(Q)echo 'SRCXX = $(SKETCH_CPP)'               > $@
```

Comment:
Is there any way this recipe can be called multiple times during the same build? (race condition clobbering the destination file)
For example if building multiple sketches in different source directories?

Reply:
You cannot put sketches in multiple directories as `post.snip` is defining the `main` function.
As I understood it, its to replace the code you would put in your application directory by this arduino sketches thing. That's why SKETCHES are in `APPDIR` which is basically `CURDIR`.